### DISCUSSION TITLE
Make E2 error on division by zero

### DISCUSSION THREAD

**Discussion by vurvdev** [wiremod/wire#2723] [Suggestions]
I see no reason for why users would want to keep a nan type, and it just leaves open potential exploits and whatnot as in https://github.com/wiremod/wire/commit/c7cfcaa15a9a02b3a79d6c19e9c7a995d218dcfb.

So I propose one of these things to happen on division by zero:

1. Throw a runtime error, regardless of `@strict`, on division by zero
2. Throw a runtime error on `@strict`, or return `0` without ([gates do this](https://github.com/wiremod/wire/blob/c7cfcaa15a9a02b3a79d6c19e9c7a995d218dcfb/lua/wire/gates/arithmetic.lua#L223))

Would need to deprecate `isnan` since it would never hold true.

**Comment by thegrb93** [comment]
Divide by 0 generates inf and 0/0 generates nan. You'd need to do this for every math function that can return error floats as well.
